internal/resource-request-operator/broadcaster.go

```diff
@@ -147,47 +170,104 @@ func (b *Broadcaster) onNodeDelete(obj interface{}) {
+				// sub the resource of this pod in the map clusterID => resources to be used in ReadResources() function.
+				// this action is done to correct the computation not considering pod offloaded by the cluster with this ClusterID
+				subResources(currentPodsResources, oldResources)
+			}
+		}
+	}
 	b.writeClusterResources(currentResources)
-	if clusterID := pod.Labels[forge.LiqoOriginClusterID]; clusterID != "" {
-		klog.V(4).Infof("OnPodAdd: Pod %s:%s passed ClusterID check. ClusterID = %s\n", pod.Namespace, pod.Name, clusterID)
-		currentPodsResources := b.readPodResources(clusterID)
-		// add the resource of this pod in the map clusterID => resources to be used in ReadResources() function.
-		// this action is done to correct the computation not considering pod offloaded by the cluster with this ClusterID
-		addResources(currentPodsResources, podResources)
+	if clusterID != "" {
 		b.writePodsResources(clusterID, currentPodsResources)
 	}
 }
 
 func (b *Broadcaster) onPodDelete(obj interface{}) {
-	pod := obj.(*corev1.Pod)
-	klog.V(4).Infof("OnPodDelete: Delete for pod %s:%s\n", pod.Namespace, pod.Name)
-	podResources := extractPodResources(pod)
-	currentResources := b.readClusterResources()
-	// this action is the complementary of the onPodAdd() one. For more details see at that function.
-	addResources(currentResources, podResources)
-	b.writeClusterResources(currentResources)
-	if clusterID := pod.Labels[forge.LiqoOriginClusterID]; clusterID != "" {
-		klog.V(4).Infof("OnPodDelete: Pod %s:%s passed ClusterID check. ClusterID = %s\n", pod.Namespace, pod.Name, clusterID)
-		currentPodsResources := b.readPodResources(clusterID)
-		subResources(currentPodsResources, podResources)
-		b.writePodsResources(clusterID, currentPodsResources)
+	podDeleted := obj.(*corev1.Pod)
+	klog.V(4).Infof("OnPodDelete: Delete for pod %s:%s\n", podDeleted.Namespace, podDeleted.Name)
+	if pod.IsPodReady(podDeleted) {
+		podResources := extractPodResources(podDeleted)
+		currentResources := b.readClusterResources()
+		// Resources used by the pod will become available again so add them to the total allocatable ones.
+		addResources(currentResources, podResources)
+		b.writeClusterResources(currentResources)
+		if clusterID := podDeleted.Labels[forge.LiqoOriginClusterID]; clusterID != "" {
+			klog.V(4).Infof("OnPodDelete: Pod %s:%s passed ClusterID check. ClusterID = %s\n", podDeleted.Namespace, podDeleted.Name, clusterID)
+			currentPodsResources := b.readPodResources(clusterID)
+			subResources(currentPodsResources, podResources)
+			b.writePodsResources(clusterID, currentPodsResources)
+		}
 	}
 }
 
 // write nodes resources in thread safe mode.
 func (b *Broadcaster) writeClusterResources(newResources corev1.ResourceList) {
+	err := checkSign(newResources)
+	if err != nil {
+		klog.Errorf("Broadcaster -> Writing global resources error: %s", err)
```

Comment:
I think that we should do something here, or default to 0 the value, or not update the stored value. What to you think @palexster ?

Reply:
Ok, my personal feeling is that we should notice immediately when a negative offer is generated, to avoid problems and bugs we had in the past. Since it is potentially a critical error and impossible to obtain without a bug in our implementation, I would make the broadcaster fail or something that cannot be ignored. What do you think? @aleoli @alacuku @giorio94

Reply:
I am not entirely sure here. Panic'ing would probably be a sensible option to make sure errors do not get unnoticed, but it would cause the restart of the entire `liqo-controller-manager`. Additionally, it could never recover, making `liqo` unusable. I feel it could make sense very short term, but should be probably reverted to an error message (possibly defaulting to 0) before the 0.3 release.